riptide-core/src/main/java/org/zalando/riptide/DefaultHttpBuilder.java

```diff
@@ -129,22 +127,23 @@ public ConfigurationStage defaultPlugins() {
 
     @Override
     public ConfigurationStage plugins(final Iterable<Plugin> plugins) {
-        return new DefaultHttpBuilder(executor, io, converters, baseUrl, resolution,
-                this.plugins.concat(plugins));
+        return withPlugins(this.plugins.concat(plugins));
     }
 
     @Override
     public ConfigurationStage plugin(final Plugin plugin) {
-        return new DefaultHttpBuilder(executor, io, converters, baseUrl, resolution,
-                plugins.append(plugin));
+        return withPlugins(plugins.append(plugin));
     }
 
     @Override
     public Http build() {
         final List<HttpMessageConverter<?>> converters = converters();
-
         final List<Plugin> plugins = new ArrayList<>();
-        plugins.add(new AsyncPlugin(executor));
+
+        if (executor != null) {
+            plugins.add(new AsyncPlugin(executor));
```

Comment:
Next step could be to remove the ability to configure an executor from the API and expose this plugin directly (e.g. a `ThreadPoolPlugin`) which could then actually be moved outside of the core (`riptide-concurrent`?), together with the content of the `org.zalando.riptide.concurrent` package.

Reply:
Leaves the question of how to call the plugin phase that is currently called `aroundAsync`.

Reply:
`aroundRiptide` :wink: